groups/nts/ntsu/ntsu_socketutil.t.cpp

```diff
@@ -616,6 +709,221 @@ void testStreamSocketTransmissionBlob(ntsa::Transport::Value transport,
+    }
+
+    // Dequeue incoming data received by the server socket.
+
+    {
+        ntsa::ReceiveContext context;
+        ntsa::ReceiveOptions options;
+
+        options.showForeignHandles();
+
+        error =
+            ntsu::SocketUtil::receive(&context, &serverBlob, options, server);
+        NTSCFG_TEST_ASSERT(!error);
+
+        NTSCFG_TEST_ASSERT(context.bytesReceivable() == 9);
+        NTSCFG_TEST_ASSERT(context.bytesReceived() == 9);
+
+        NTSCFG_TEST_ASSERT(serverBlob.length() == 9);
+        NTSCFG_TEST_ASSERT(bdlbb::BlobUtil::compare(serverBlob, clientBlob) ==
+                           0);
+
+        NTSCFG_TEST_TRUE(!context.foreignHandle().isNull());
+
+        ntsa::Endpoint foreignSourceEndpoint;
+        error = ntsu::SocketUtil::sourceEndpoint(
+            &foreignSourceEndpoint, 
+            context.foreignHandle().value());
+        NTSCFG_TEST_ASSERT(!error);
+
+        NTSCFG_TEST_EQ(foreignSourceEndpoint, domesticSourceEndpoint);
+
+        error = ntsu::SocketUtil::close(domesticSocket);
+        NTSCFG_TEST_ASSERT(!error);
+
+        error = ntsu::SocketUtil::close(context.foreignHandle().value());
+        NTSCFG_TEST_ASSERT(!error);
+    }
+}
+
+void testStreamSocketTransmissionWithControlMsgDropped(
+                                      ntsa::Transport::Value transport,
+                                      ntsa::Handle           server,
+                                      ntsa::Handle           client,
+                                      bslma::Allocator*      allocator)
+{
+    if (transport != ntsa::Transport::e_LOCAL_STREAM) {
+        return;
+    }
+
+    NTSCFG_TEST_LOG_DEBUG << "Testing " << transport 
+                          << ": writev/readv (blob) "
+                          << "with ancillary data"
+                          << NTSCFG_TEST_LOG_END;
+
+    ntsa::Error error;
+
+    char DATA[] = "123456789";
+
+    bdlbb::SimpleBlobBufferFactory blobBufferFactory(3, allocator);
+
+    ntsa::Handle domesticSocket;
+    error = ntsu::SocketUtil::create(&domesticSocket, transport);
+    NTSCFG_TEST_ASSERT(!error);
+
+    error = ntsu::SocketUtil::bind(
+        ntsa::Endpoint(ntsa::LocalName::generateUnique()),
+        false,
+        domesticSocket);
+    NTSCFG_TEST_ASSERT(!error);
+
+    ntsa::Endpoint domesticSourceEndpoint;
+    error = ntsu::SocketUtil::sourceEndpoint(&domesticSourceEndpoint, 
+                                             domesticSocket);
+    NTSCFG_TEST_ASSERT(!error);
+
+    for (bsl::size_t iteration = 0; iteration < 2; ++iteration) {
+        NTSCFG_TEST_LOG_DEBUG << "Testing iteration " << iteration 
+                              << NTSCFG_TEST_LOG_END;
+
+        bdlbb::Blob clientBlob(&blobBufferFactory, allocator);
+        bdlbb::BlobUtil::append(&clientBlob, DATA, sizeof DATA - 1);
+
+        bdlbb::Blob serverBlob(&blobBufferFactory, allocator);
+        serverBlob.setLength(sizeof DATA - 1);
```

Comment:
It seems to be not needed. The next line sets blob's length to 0. I see the same pattern in other tests where `bdlbb::Blob` is used.

Reply:
This is the idiom for `bdlbb::Blob` to `reserve(<size>)`.